termwiz/src/color.rs

```diff
@@ -71,31 +69,34 @@ impl RgbColor {
         Self { red, green, blue }
     }
 
-    pub fn to_linear(self) -> LinSrgba {
-        Srgba::<u8>::new(self.red, self.green, self.blue, 0xff)
-            .into_format()
-            .into_linear()
-    }
-
     pub fn to_tuple_rgba(self) -> RgbaTuple {
-        Srgba::<u8>::new(self.red, self.green, self.blue, 0xff)
-            .into_format()
-            .into_components()
-    }
-
-    pub fn to_linear_tuple_rgba(self) -> RgbaTuple {
-        self.to_linear().into_components()
+        (
+            self.red as f32 / 255.0,
+            self.green as f32 / 255.0,
+            self.blue as f32 / 255.0,
+            1.0,
+        )
     }
 
     /// Construct a color from an SVG/CSS3 color name.
-    /// Returns None if the supplied name is not recognized.
+    /// Returns None if the supplied name is not recognized, or the
+    /// `color-names` feature is disabled.
+    ///
     /// The list of names can be found here:
     /// <https://ogeon.github.io/docs/palette/master/palette/named/index.html>
     pub fn from_named(name: &str) -> Option<RgbColor> {
-        palette::named::from_str(&name.to_ascii_lowercase()).map(|color| {
-            let color = Srgb::<u8>::from_format(color);
-            Self::new(color.red, color.green, color.blue)
-        })
+        #[cfg(feature = "color-names")]
```

Comment:
I think I'd like to replace this with a lazy-static hashmap initialized from rgb.txt.
That file comes originally from X11, but vim also has a decent copy:

```
$ wc -l /usr/share/vim/vim82/rgb.txt
782 /usr/share/vim/vim82/rgb.txt
$ head /usr/share/vim/vim82/rgb.txt
255 250 250             snow
248 248 255             ghost white
248 248 255             GhostWhite
245 245 245             white smoke
245 245 245             WhiteSmoke
220 220 220             gainsboro
255 250 240             floral white
255 250 240             FloralWhite
253 245 230             old lace
253 245 230             OldLace
```

it should hopefully be pretty trivial to `include_str!` that file and build up a map of the lowercased name -> rgb values for use here

Reply:
I added an rgb.txt based alternative to the palette crate for this case, so we shouldn't need the color names feature in this PR